avocado/core/runner.py

```diff
@@ -468,6 +469,12 @@ def _iter_variants(template, mux):
         :raises ValueError: When variant and template declare params.
         """
         for variant, params in mux.itertests():
+            variant_name = ''
+            for x in params[0]:
+                path = x.path[len('/run'):]
+                variant_name += path.replace('/', '-')
+            # trim first '-'
+            variant_name = variant_name[1:]
```

Comment:
(Basically instead of changing the concept, I'd prefer to define `variant-id` as something more than just a string...)